### ISSUE TITLE
Bring in signal code from OWS's current client

### ISSUE THREAD

**Issue by johnhaley81** [johnhaley81/signal-desktop#3]
Just to get something going let's get the code from https://github.com/WhisperSystems/Signal-Desktop in here and then start replacing stuff.

**Comment by johnhaley81** [comment]
@srajko we could probably use https://github.com/koush/electron-chrome to bring it in.

**Comment by srajko** [comment]
Yep, I looked at that this morning.  After I mentioned to you that there might be value in making a general-purpose chrome app compatibility layer, I figured there might already be one :-)

Signal doesn't run out of the box with https://github.com/koush/electron-chrome (it starts to launch but crashes on a JavaScript error), so there are probably some gaps to fill.  That's probably step one, if we get it working we might want to extract the polyfill parts of electron-chrome into an npm package, then use that from here.

**Comment by srajko** [comment]
In terms of pulling in https://github.com/WhisperSystems/Signal-Desktop and then replacing stuff - if the chrome app compatibility layer pans out, we could also keep this project tied to Signal-Desktop and make contributions there instead (rather than forking the Signal-Desktop code).

**Comment by srajko** [comment]
I made a local fork https://github.com/signal-desktop/electron-chrome so we can add whatever we need to get signal running.

**Comment by johnhaley81** [comment]
I like the idea of bridging the 2 projects instead of just going off in a different direction.

**Comment by maxkorp** [comment]
I mentioned this in person, but I've already got most of the libtextsecure from their desktop app refactored. I'll try to get it up tonight, probably as an additional project

**Comment by kfatehi** [comment]
Hey guys -- this repo was on my radar because I was looking into using this to build a [Matrix](https://matrix.org) bridge, but couldn't find the refactored libtextsecure code. FWIW I ended up doing a dirty port of the chrome app to node.js, which can be found here https://github.com/matrix-hacks/matrix-puppet-signal -- maybe it's useful to you.
